### PR TITLE
✨ Add support for Cadence struct to Go struct

### PR DESCRIPTION
**Pull request by LemonNekoGH**

### CODE REVIEW

**Comment by ritz078** [comment]
#### 📝 Changed routes:


| [**/cadence-to-go**](https://transform-wdiymg037-ritz078.vercel.app/cadence-to-go) |
|:-:|
|<a href="https://transform-wdiymg037-ritz078.vercel.app/cadence-to-go">Screenshot of /cadence-to-go</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/7cspxp7ygZOp1Y3SMsxjfcyj5uHArcKriSBYVhsGr3vMmHBh0wvAX7BNaYYWiQG1qpvy_EJmsCIiWxo1-vPmsg.png">(view full size)</a> |







Commit ca26a260398c1ceb0fe698ab919b5dbf0f06d654 (https://transform-wdiymg037-ritz078.vercel.app).

File: pages/cadence-to-go.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+export default function CadenceToGo() {
+  const transformer = useCallback(({ value }) => transform(value),[]);
+
+  return (
+    <ConversionPanel
+      transformer={transformer}
+      editorTitle="Cadence"
+      editorLanguage="cadence"
```
Comment (ritz078) [inline pages/cadence-to-go.tsx:74]: Cadence is not supported by Monaco editor so you should change this to "text"

File: pages/cadence-to-go.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+async function cadenceToGo(value: string): Promise<string> {
+  const parser = await CadenceParser.create("https://cdn.jsdelivr.net/npm/@lemonneko/flow__cadence-parser@0.24.1/dist/cadence-parser.wasm")
+  const ast = parser.parse(value);
+  const program = ast.program
+  const declarations = program.Declarations
+  let ret = ''
+  for (const dec of declarations) {
+    console.log(dec)
```
Comment (ritz078) [inline pages/cadence-to-go.tsx:43]:
```suggestion
```

File: pages/cadence-to-go.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      for (const member of dec.Members.Declarations) {
+        if (member.Type === 'FieldDeclaration') {
+          decRet += `${firstLetterUppercase(member.Identifier.Identifier)} ${typeMap[member.TypeAnnotation.AnnotatedType.Identifier.Identifier]} \`godence:"${member.Identifier.Identifier}"\`\n`
+        }
+      }
+      decRet += '\n}'
+    }
+    ret += decRet
+  }
+  console.log(ret)
```
Comment (ritz078) [inline pages/cadence-to-go.tsx:57]:
```suggestion
```

**Review by ritz078 — CHANGES_REQUESTED** [review]
I have added a few comments for your consideration. Thank you.

**Comment by ritz078** [comment]
Sorry for the (very) late response. Thank you for your contribution 🎉

**Review by ritz078 — APPROVED** [review]

**Comment by ritz078** [comment]
@allcontributors please add @LemonNekoGH for code.

**Comment by LemonNekoGH** [comment]
Thank you!

**Pull request merged**